hail_search/hail_search_query.py

```diff
@@ -677,29 +681,59 @@ def _format_transcript_args(self):
 
     def _get_family_passes_quality_filter(self, quality_filter, ht=None, pathogenicity=None, **kwargs):
         passes_quality = super(VariantHailTableQuery, self)._get_family_passes_quality_filter(quality_filter)
-        clinvar_path_ht = False if passes_quality is None else self._get_clinvar_filter_ht(pathogenicity)
+        clinvar_path_ht = False if passes_quality is None else self._get_loaded_filter_ht(
+            CLINVAR_KEY, 'clinvar_path_variants.ht', self._get_clinvar_prefilter, pathogenicity=pathogenicity)
         if not clinvar_path_ht:
             return passes_quality
 
         return lambda entries: hl.is_defined(clinvar_path_ht[ht.key]) | passes_quality(entries)
 
-    def _get_clinvar_filter_ht(self, pathogenicity):
-        if self._filter_hts.get(CLINVAR_KEY) is not None:
-            return self._filter_hts[CLINVAR_KEY]
+    def _get_loaded_filter_ht(self, key, table_path, get_filters, **kwargs):
+        if self._filter_hts.get(key) is not None:
+            return self._filter_hts[key]
+
+        ht_filter = get_filters(**kwargs)
+        if ht_filter is False:
+            self._filter_hts[key] = False
+            return False
+
+        ht = self._read_table(table_path)
+        if ht_filter is not True:
+            ht = ht.filter(ht[ht_filter])
+        self._filter_hts[key] = ht
 
+        return ht
+
+    def _get_clinvar_prefilter(self, pathogenicity=None):
         clinvar_path_filters = self._get_clinvar_path_filters(pathogenicity)
         if not clinvar_path_filters:
-            self._filter_hts[CLINVAR_KEY] = False
             return False
 
-        clinvar_path_ht = self._read_table('clinvar_path_variants.ht')
         if CLINVAR_LIKELY_PATH_FILTER not in clinvar_path_filters:
-            clinvar_path_ht = clinvar_path_ht.filter(clinvar_path_ht.is_pathogenic)
+            return 'is_pathogenic'
         elif CLINVAR_PATH_FILTER not in clinvar_path_filters:
-            clinvar_path_ht = clinvar_path_ht.filter(clinvar_path_ht.is_likely_pathogenic)
-        self._filter_hts[CLINVAR_KEY] = clinvar_path_ht
+            return 'is_likely_pathogenic'
+        return True
+
+    def _prefilter_entries_table(self, ht, **kwargs):
+        af_ht = self._get_loaded_filter_ht(
+            GNOMAD_GENOMES_FIELD, 'high_af_variants.ht', self._get_gnomad_af_prefilter, **kwargs)
```

Comment:
The function `_prefilter_entries_table` looks prefiltering for `gnomAD`. How about adding `gnomad` to the function name?

Reply:
this is a subclass specific override of a base method. For the variants table it filters for gnomad, for other data types it will have different filtering